runtime/gc_modron_startup/mgcalloc.cpp

```diff
@@ -270,7 +271,8 @@ traceObjectCheck(J9VMThread *vmThread)
 	if(extensions->doOutOfLineAllocationTrace){
 		uintptr_t byteGranularity = extensions->oolObjectSamplingBytesGranularity;
 	
-		if(env->_oolTraceAllocationBytes >= byteGranularity){
+		/* if we want to trace ool Allocation, use env->_oolTraceAllocationBytes for comparison */
+		if(env->_traceAllocationBytes >= byteGranularity){
```

Comment:
Please correct me if I missed something: We are trying to implement new tracing mode using `_traceAllocationBytes` however need to support existing tracing using `_oolTraceAllocationBytes` (mutually exclusive cases obviously) sharing `oolObjectSamplingBytesGranularity` for sampling interval. If it is a case should we have both checks here (is it possible to combine)? Would existing tracing be broken by this change?

Reply:
So far I know only health centre use it, I think that using _traceAllocationBytes would help health centre to get more accurate report too, if we want to keep existing tracing, we need to vm side to update to split usage. @JasonFengJ9 any comment?

Reply:
Also we do increment `_oolTraceAllocationBytes` in `MM_TLHAllocationInterface::allocateObject()` but can not find any similar place for `_traceAllocationBytes`. Where it is incremented?

Reply:
@dmitripivkine the change at line 206 of MM_TLHAllocationInterface::allocateObject(), the code had merged week ago.
```
	if (NULL != result) {
		uintptr_t sizeInBytesAllocated = allocDescription->getContiguousBytes();
		/* Increment by bytes allocated */
		env->_traceAllocationBytes += sizeInBytesAllocated;

		if (!allocDescription->isCompletedFromTlh()) {
#if defined(OMR_GC_OBJECT_ALLOCATION_NOTIFY)
			env->objectAllocationNotify((omrobjectptr_t)result);
#endif /* OMR_GC_OBJECT_ALLOCATION_NOTIFY */
			_stats._allocationBytes += sizeInBytesAllocated;
			_stats._allocationCount += 1;
		}
	}
	env->_oolTraceAllocationBytes += (_stats.bytesAllocated() - _bytesAllocatedBase); /* Increment by bytes allocated */

```

Reply:
Thank you. I looked to outdated code, sorry

Reply:
> So far I know only health centre use it, I think that using _traceAllocationBytes would help health centre to get more accurate report too, if we want to keep existing tracing, we need to vm side to update to split usage.

If new feature covers old functionality (do we have tests for it?), it is ok for me to have new one only. I was confused by your change in `traceAllocateObject()` where you seems try to maintain both variables (why?)
If it is a case more clearing/renaming is required - should `_oolTraceAllocationBytes` be removed and `_ool` prefix variables be renamed?